src/lib/tvgPictureImpl.h

```diff
@@ -220,6 +258,20 @@ struct Picture::Impl
         return Result::Success;
     }
 
+    bool mesh(const Polygon* triangles, const uint32_t triangleCount)
+    {
+        if (triangles && triangleCount > 0) {
+            this->triangleCount = triangleCount;
+            this->triangles = (Polygon*)malloc(sizeof(Polygon) * triangleCount);
+            for (uint32_t i = 0; i < triangleCount; i++) this->triangles[i] = triangles[i];
+        } else {
+            free(this->triangles);
+            this->triangles = nullptr;
+            this->triangleCount = 0;
+        }
+        return true;
+    }
+
     Paint* duplicate()
```

Comment:
I missed one point, duplicate() must copy the interal member data of the Picture. So do polygon data as well.
Could you please copy mesh data here as well?